dbms/src/Storages/DeltaMerge/Filter/ColumnRange.h

```diff
@@ -0,0 +1,193 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Storages/DeltaMerge/Filter/IntegerSet.h>
+#include <Storages/KVStore/Types.h>
+
+#include <magic_enum.hpp>
+
+namespace DB::DM
+{
+
+using IndexID = DB::IndexID;
+using ColumnID = DB::ColumnID;
+
+class ColumnRange;
+using ColumnRangePtr = std::shared_ptr<ColumnRange>;
+using ColumnRanges = std::vector<ColumnRangePtr>;
+
+class SingleColumnRange;
+using SingleColumnRangePtr = std::shared_ptr<SingleColumnRange>;
+
+enum class ColumnRangeType
+{
+    Unsupported,
+    Single,
+    And,
+    Or,
+};
+
+// ColumnRange represents the range of values for columns.
+class ColumnRange : public std::enable_shared_from_this<ColumnRange>
+{
+public:
+    explicit ColumnRange(ColumnRangeType type_)
+        : type(type_)
+    {}
+
+    virtual ~ColumnRange() = default;
+
+    virtual ColumnRangePtr invert() const = 0;
+
+    virtual String toDebugString() = 0;
+
+    virtual BitmapFilterPtr check(std::function<BitmapFilterPtr(const ColumnRangePtr &, size_t)> search, size_t size)
+        = 0;
```

Comment:
Add some comments about the param of the function `check` and its params

Reply:
Do this in Next PR https://github.com/pingcap/tiflash/pull/10006.